### DISCUSSION TITLE
Pass argument to ${command:pickFile}

### DISCUSSION THREAD

**Discussion by Am1n3e** [mfussenegger/nvim-dap#1399] [General]
Hello, 

Is there a way to pass arguments to `${command:pickFile}` like the path from `launch.json`?

Thanks!

**Comment by mfussenegger** [comment]
Currently there's no option to pass arguments to it. Does vscode offer functionality for that?

Alternative option is to define the configuration in Lua, or you could use a `on_config` hook to create your own `${command:pickFile-xy}` variants. See `:help dap-listeners-on_config`. See https://github.com/mfussenegger/nvim-dap/issues/1360#issuecomment-2473987283 for examples

**Reply by Am1n3e** [reply]
@mfussenegger Thanks for the quick response.

I've been using the [Command Varriable](https://marketplace.visualstudio.com/items?itemName=rioj7.command-variable#pick-file) plug-in that I think is very useful when you need to select files for the cmd args.  It looks like the following:

```json
  "data_config_picker": {
      "display": "relativePath",
      "description": "Config",
      "fromFolder": {
          "fixed": "${workspaceFolder}/config/data"
      },
      "canPickMany": false,
      "include": "**/*.yaml",
      "exclude": "**/_*.*",
      "transform": {
          "text": "${relativeFile}",
          "find": "config/",
          "replace": "",
      }
  },
```

In my use case, I need to select 1 to 3 config files as args. So, I used to rely on it to show a file picker for each arg (at different locations). 
It also can store the previously used combination to use in a later session.

**Reply by Am1n3e** [reply]
Forgot the mention the configuration above, is added the "inputs" in `launch.json`. Similar to `promptString`.

**Reply by mfussenegger** [reply]
I see. Given that it requires a plugin in vscode it's definitely out-of-scope for nvim-dap core, but you should be able to use the `on_config` mechanism to add support for it.

**Comment by Am1n3e** [comment]
Of course one can add that the nvim config, however, that limits sharing when others working on same code repository rely on VsCode.

**Comment by Am1n3e** [comment]
@mfussenegger do you think that could be possible to have some simple version of what I showed above add to the `nvim-dap`. 

A simple MVP would be able to set the following from `launch.json`
- The base path for the fuzzy finder.
- The prompt 
- The filters
Also it will nice to have the ability to select more than one file and the list of file joined by a given string.

```json
 
{
  "configurations": [
    {
      "name": "Run",
      "type": "debugpy",
      "request": "launch",
      "program": "run.py",
      "console": "internalConsole",
      "args": [
        "--arg1=${input:cfg_1_picker}",
        "--arg2=${input:cfg_2_picker}",
      ]
    }
  ],
  "inputs": [
    {
      "id": "cfg_1_picker",
      "type": "command",
      "command": "pickFile",
      "args": {
        "prompt": "Select cfg 1",
        "path": "/path/cfg1/",
        "filters": "*.yaml"
      }
    }
  ]
  ...
}
```

```

**Reply by Am1n3e** [reply]
Nvm, I've read your other comment. Thanks for the suggestion I will give it a try.
